phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionlessQueryServicesImpl.java

```diff
@@ -289,8 +289,11 @@ public MetaDataMutationResult getTable(PName tenantId, byte[] schemaBytes, byte[
     }
     
     private static List<HRegionLocation> generateRegionLocations(byte[] physicalName, byte[][] splits) {
-        byte[] startKey = HConstants.EMPTY_START_ROW;
         List<HRegionLocation> regions = Lists.newArrayListWithExpectedSize(splits.length);
+        if (splits.length == 0) {
+            return regions;
+        }
+        byte[] startKey = HConstants.EMPTY_START_ROW;
```

Comment:
nit:
```suggestion
        if (splits.length == 0) {
            return regions;
        }
        List<HRegionLocation> regions = Lists.newArrayListWithExpectedSize(splits.length);
        byte[] startKey = HConstants.EMPTY_START_ROW;
```

Reply:
This won't compile, we need an empty List to return.